packages/calcite-components/src/components/color-picker/color-picker.tsx

```diff
@@ -1376,13 +1374,19 @@ export class ColorPicker
     );
   };
 
-  private initColorField = (canvas: HTMLCanvasElement): void => {
+  private initColorField = (canvas?: HTMLCanvasElement): void => {
+    if (!canvas) {
```

Comment:
Two important differences to note about `ref` props:
- If component is disconnected, in Stencil `ref` callback is not called with null/undefined. `ref` callback is only called with `null` if the component re-renders without the VNode for the ref element. In Lit/React/Preact, if component is disconnected, all ref callbacks inside of it will be called with `undefined`.
- In **Lit**, and in **React**/**Preact** if to a `ref` prop you pass a new arrow function on each render, both frameworks will call the arrow function instance from previous render with `undefined` (Lit) or `null` (React/Preact), and then call the new arrow function with the element reference.
  
   In **Stencil**, changes to the ref callback do not make it be called again. The `ref` callback is only called again when the element is actually disconnected.

Reply:
In practice that means that ref callbacks should always guard against `ref` callback being called with a nullish value.
Calcite does not have TypeScript strictNullChecks option enabled. If it did, then TypeScript would detect cases where you forgot to handle the nullish ref case, preventing a bug from happening.